core/domain/state_domain.py

```diff
@@ -159,30 +160,46 @@ def get_all_html_content_strings(self):
 
         outcome_html = self.outcome.feedback.html
         html_list = html_list + [outcome_html]
+        # NOTE TO DEVELOPERS:
+        # The rules_index_dict below is used to  igure out the assembly of the
+        # html in the rule specs.
+        # See issue: https://github.com/oppia/oppia/issues/9413. We cannot use
+        # the interaction-id from the rules_index_dict dict until issue-9413
+        # has been fixed, because this method has no reference to the
+        # interaction type.
+
+        rules_index_dict = json.loads(
+            utils.get_file_contents(feconf.RULES_DESCRIPTIONS_FILE_PATH))
+
+        set_of_html_strings_rules = (
```

Comment:
Add an additional key to the dicts in the new file saying something like `format: 'list'` or `format: 'string'`. Then here, have something like: if format == 'list': ... elif format == 'string': ... else raise Exception. 

Basically, this method needs to be as generic as possible.

Reply:
Here we are checking the type of the `rule_specs.inputs['x']` because the rule type `Equals`(which belongs to `SetOfHtmlString`) is used by other interactions as well(we are interested in only `ItemSelectionInput` ) and we don't have a reference to the interaction ID(this is the only way to differentiate these interactions which use the rule type `Equals`).
Adding the format in the new file won't help here because we have to check the type of rule_specs.inputs['x'] anyway.

Reply:
done